### ISSUE TITLE
`OutdatedDocumentation` false positive with non-public constructor params

### ISSUE THREAD

**Issue by osipxd** [detekt/detekt#5217]
## Expected Behavior
`@param` is not required for non-public parameters

## Observed Behavior
`@param` or `@property` is required for all parameters if:
- both constructor and parameter are not public
- constructor contains at least one public property

## Steps to Reproduce

Run detekt with OutdatedDocumentation rule enabled on this code

```kotlin
/**
 * Exception thrown from MyLibrary that can not be created outside of the library.
 * @property errorCode Blah-blah
 */
public class MyLibraryException internal constructor(
    public val errorCode: String,
    cause: Throwable?,
) : Exception(cause)
```
You will get `OutdatedDocumentation` for `MyLibraryException`. But if you add `@param cause` to KDoc, check will pass.

## Your Environment

* Version of detekt used: 1.20.1
* Version of Gradle used (if applicable): 7.5.1
* Gradle scan link (add `--scan` option when running the gradle task): -
* Operating System and version: macOS 12.4
* Link to your project (if it's a public repository): -
